docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,139 @@
+---
+title: api-for-skipping-validations
+authors:
+  - "@pmaidmen"
+creation-date: 2023-01-08
+last-updated: 2023-01-08
+---
+
+# Allow validations to be skipped
+
+# Summary
+
+We need new api that will allow us to skip cluster/host validations.
+This api should have it's own feature flag to allow us to disable it.
+
+It should be possible to use this API in an unsupported way, it is currently undecided whether or not this will be open or not.
+An event should be emitted whenever the API is used during an installation, this can be used to drive signatures etc.
+# Motivation
+
+Some customer and partners have very specific HW that doesn't pass our validations and we want to allow them to proceed with installation.
+Sometimes we have bugs in our validations that block people from installing and we don't want our partners left in a position where they can't install.
+
+### Goals
+
+- Allow validations to be bypassed so that installation may proceed.
+
+### Non-Goals
+
+- We should not add uneccesary noise to triage tickets with this, it is therefore quite important to add some signatures to indicate that these validations have been bypassed.
+
+## Proposal
+
+I think the API should be kept reasonably simple.
+
+### List existing validations.
+
+GET api/v2/skippedValidations
+Will return a JSON array of all the validations that are currently being skipped, format similar to
+
+```
+{
+  "skipped-host-validations": [
+    "no-ip-collisions-in-network",
+    "sufficient-network-latency-requirement-for-role'"
+  ],
+  "skipped-cluster-validations": [
+    "network-prefix-valid",
+    "api-vips-defined"
+  ]
+}
+```
+
+### Add some skipped validations.
+
+POST api/v2/skippedValidations
```

Comment:
Would it be simpler to add single validation to skip in the operation?

Reply:
I personally think this would be fine if this request is "additive", i.e. if you call it again with a different validation id then this will additionally be skipped, how does this sound?